src/oscar/apps/checkout/surcharges.py

```diff
@@ -27,11 +27,20 @@ def __init__(self, percentage):
         self.percentage = percentage
 
     def calculate(self, basket, **kwargs):
-        if basket.total_excl_tax:
+        if not basket.is_empty:
+            shipping_charge = kwargs.get("shipping_charge", None)
```

Comment:
A default of None is obsolete, because that is the default allready.

Reply:
Fixed